File: rust/examples/test_mixed.rs
use emtrace::{Out, expect, init, traceln};

fn main() {
    expect!(
        "Mixed types: 42 3.140000104904175 True hello\n\
          Complex format: Value=100, Name=test, Active=False, Ratio=0.5\n\
          Numbers: \x01 \x02 \x03 \x04 \x05\n"
    );

    init(&mut std::io::stdout().lock());

    // Test mixing different types in one trace
    traceln!("Mixed types: {} {} {} {}",
        i32: 42,
        f32: 3.140000104904175f32,
        bool: true,
        str: "hello"
    );

    // Test more complex formatting
    traceln!("Complex format: Value={}, Name={}, Active={}, Ratio={}",
        i32: 100,
        str: "test",
        bool: false,
        f64: 0.5
    );

    // Test multiple same-type values (u8 shows as characters, not numbers)
    traceln!("Numbers: {} {} {} {} {}",
        u8: 1u8,
        u8: 2u8,
        u8: 3u8,
        u8: 4u8,
        u8: 5u8
    );
}

File: c/test/test_mixed.c
#include "test_utils.h"
#include <emtrace/emtrace.h>
#include <stdbool.h>
#include <stdint.h>

EXPECT_OUTPUT(
    "Mixed types: 42 3.140000104904175 True\n"
    "Complex format: Value=100, Active=False, Ratio=0.5\n"
    "Numbers: \x01 \x02 \x03 \x04 \x05\n"
);

int main(void) {
    EMTRACE_INIT();

    // Test mixing different types in one trace
    int int_val = 42;
    float float_val = 3.140000104904175F; // Use exact expected value
    bool bool_val = true;

    EMTRACELN_F("Mixed types: {} {} {}", int, int_val, float, float_val, bool, bool_val);

    // Test more complex formatting
    int complex_int = 100;
    bool complex_bool = false;
    double complex_double = 0.5;

    EMTRACELN_F(
        "Complex format: Value={}, Active={}, Ratio={}", int, complex_int, bool, complex_bool,
        double, complex_double
    );

    // Test multiple same-type values (uint8_t shows as characters, not numbers)
    uint8_t n1 = 1;
    uint8_t n2 = 2;
    uint8_t n3 = 3;
    uint8_t n4 = 4;
    uint8_t n5 = 5;

    EMTRACELN_F(
        "Numbers: {} {} {} {} {}", uint8_t, n1, uint8_t, n2, uint8_t, n3, uint8_t, n4, uint8_t, n5
    );

    return 0;
}
